cfgov/ask_cfpb/models/pages.py

```diff
@@ -241,6 +244,210 @@ def page_js(self):
+    """
+    A routable page type for Ask CFPB portal search ("see-all") pages.
+    """
+
+    objects = CFGOVPageManager()
+    portal_topic = models.ForeignKey(
+        PortalTopic,
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL)
+    portal_category = None
+    overview = models.TextField(blank=True)
+    content_panels = CFGOVPage.content_panels + [
+        FieldPanel('portal_topic'),
+        FieldPanel('overview'),
+    ]
+    edit_handler = TabbedInterface([
+        ObjectList(content_panels, heading='Content'),
+        ObjectList(CFGOVPage.settings_panels, heading='Configuration'),
+    ])
+
+    @property
+    def query_base(self):
+        return SearchQuerySet().filter(
+            portal_topics=self.portal_topic.heading,
+            language=self.language)
+
+    @property
+    def category_map(self):
+        return {
+            slugify(_(category.heading)): category
+            for category in PortalCategory.objects.all()
+        }
+
+    def results_message(self, count, heading, search_term):
+        if search_term:
+            for_term = '{} "{}"'.format(_('for'), search_term)
+        else:
+            for_term = ''
+        if count == 1:
+            result_term = _('result')
+        else:
+            result_term = _('results')
+        if self.portal_category and search_term:
+            return format_html(
+                '{} {} {} {} {} {}<br><p class="m-notification_message_link">'
+                '<a href="../?search_term={}">'
+                '{} {}</a></p>',
+                _('Showing'),
+                count,
+                result_term,
+                for_term,
+                _('within'),
+                heading.lower(),
+                search_term,
+                _('See all results within'),
+                _(self.portal_topic.heading).lower()
+            )
+        elif self.portal_category:
+            return '{} {} {} {} {}'.format(
+                _('Showing'),
+                count,
+                _(result_term),
+                _('within'),
+                heading.lower()
+            )
+        return '{} {} {} {} {} {}'.format(
+            _('Showing'),
+            count,
+            result_term,
+            for_term,
+            _('within'),
+            heading.lower())
+
+    def get_template(self, request):
+        return 'ask-cfpb/see-all.html'
+
+    def get_heading(self, obj):
+        if self.language == 'es':
+            return obj.heading_es
+        else:
+            return obj.heading
+
+    def get_context(self, request, *args, **kwargs):
+        if self.language != 'en':
+            activate(self.language)
+        else:
+            deactivate_all()
+        return super(PortalSearchPage, self).get_context(
+            request, *args, **kwargs)
+
+    def get_nav_items(self, request, page):
+        """Return nav items sorted by an arbitrary order."""
+        pk_sort_order = [1, 4, 5, 2, 3]
```

Comment:
If these primary keys are going to be hardcoded in code like this, dependent on the fixture file, what is the advantage of using a PortalCategory model versus just defining a hardcoded Python module that defines the categories?

Reply:
@chosak I'm wondering what the best setup would be If, hypothetically, the arbitrary order were changed or we were suddenly allowed to sort alphabetically.

Reply:
Is your concern that those changes would require a code change and redeploy? Wouldn't that still be required in this case?

Reply:
True that code change is inevitable. I was aiming to have an ideal structure in place if we were freed from arbitrary order. And to me the ideal is a category lineup that can be managed by an editor without deployments. That was one goal of PortalCategory and PortalTopic.

Maybe adding an editable sort-order field to PortalCategory could accomplish both?

Reply:
I like the idea of adding a sort field to PortalCategory, if it's a must that they're sorted in a particular way.